### ISSUE TITLE
Bug rendering tree selection

### ISSUE THREAD

**Issue by SavageDD** [ocornut/imgui#2579]
**Version/Branch of Dear ImGui:**

Version: 1.71
Branch: master

**Back-end/Renderer/Compiler/OS**

Back-ends: imgui_impl_dx9.cpp + imgui_impl_win32.cpp
Operating System: Windows 10

**Screenshots/Video**

![Iea5QLs](https://user-images.githubusercontent.com/25623589/58344463-694a1880-7e5e-11e9-9447-bd39f54f4ca5.png)

**Standalone, minimal, complete and verifiable example:** _(see https://github.com/ocornut/imgui/issues/2261)_
```
main.cpp:
/////////
ImGui::NewFrame();
CMenu::Render();
ImGui::EndFrame();
///////////






static int tr = 0;
void node_bug_test() {
    
    if (ImGui::TreeNodeEx((void*)(intptr_t)tr, ImGuiTreeNodeFlags_OpenOnArrow | ImGuiTreeNodeFlags_OpenOnDoubleClick, "Scroll %i", tr)) {
        tr++;
        if (tr < 10)
            node_bug_text();
        ImGui::TreePop();
    }
}
void CMenu::Render() {
    ImGui::GetStyle().WindowRounding = 0.0f;
    ImGui::Begin("Scene", 0, ImGuiWindowFlags_None | ImGuiWindowFlags_NoCollapse | ImGuiWindowFlags_HorizontalScrollbar);
    ImGui::ShowDemoWindow();
    //node->Draw();
    node_bug_test();
    tr = 0;
    ImGui::End();
}
```

**Sorry if it's not bug**

**Comment by ocornut** [comment]
Thank you!

I think this is the same as #2211 so I'll close this as duplicate, but this is a good reminder this needs probably need solving. I'll look into it since I have more of my mind into the mess that is the ContentsRegion system + horizontal scrolling.

(PS: Please use github attachments instead of imgur, as instructed in the Issue creating guidelines)

**Comment by ocornut** [comment]
Closing as fixed, see #2211 for details.
